include/os/linux/kernel/linux/mod_compat.h

```diff
@@ -134,12 +134,14 @@ enum scope_prefix_types {
  *   spa_slop_shift
  */
 /* BEGIN CSTYLED */
-#define ZFS_MODULE_PARAM_CALL(scope_prefix, name_prefix, name, setfunc, getfunc, perm, desc) \
+#define	ZFS_MODULE_PARAM_CALL(scope_prefix, name_prefix, name, setfunc, getfunc, perm, desc) \
 	CTASSERT_GLOBAL((sizeof (scope_prefix) == sizeof (enum scope_prefix_types))); \
 	module_param_call(name_prefix ## name, setfunc, getfunc, &name_prefix ## name, perm); \
 	MODULE_PARM_DESC(name_prefix ## name, desc)
 /* END CSTYLED */
 
+#define	ZFS_MODULE_PARAM_ARGS	const char *buf, zfs_kernel_param_t *kp
```

Comment:
What does the FreeBSD version of this look like?  What args does it expect?  It would be nice if we could keep this function parameterized rather than grouping all the args together like this.

Reply:
The FreeBSD version is
```
#define	ZFS_MODULE_PARAM_ARGS	SYSCTL_HANDLER_ARGS
```
so from our perspective this is conventional for sysctl handlers.
In sys/sysctl.h SYSCTL_HANDLER_ARGS is defined as
```
#define SYSCTL_HANDLER_ARGS struct sysctl_oid *oidp, void *arg1,        \
        intmax_t arg2, struct sysctl_req *req
```
I would prefer defining the parameter list as it makes the implementation very straightforward, and it is idiomatic on FreeBSD and macOS at least.

Reply:
Huh, that's interesting.  Alright, then I'm OK with this.